File: authenticator-archetype/src/main/resources/archetype-resources/src/main/java/descriptor/__pluginName__AuthenticatorPluginDescriptor.java
package ${package}.descriptor;

import ${package}.authentication.${pluginName}AuthenticatorRequestHandler;
import ${package}.config.${pluginName}AuthenticatorPluginConfig;
import se.curity.identityserver.sdk.authentication.AuthenticatorRequestHandler;
import se.curity.identityserver.sdk.plugin.descriptor.AuthenticatorPluginDescriptor;

import java.util.Map;

import static java.util.Collections.singletonMap;
import static java.util.Collections.unmodifiableMap;

public final class ${pluginName}AuthenticatorPluginDescriptor implements AuthenticatorPluginDescriptor<${pluginName}AuthenticatorPluginConfig>
{
    @Override
    public String getPluginImplementationType()
    {
        return "${pluginName.toLowerCase()}";
    }

    @Override
    public Class<? extends ${pluginName}AuthenticatorPluginConfig> getConfigurationType()
    {
        return ${pluginName}AuthenticatorPluginConfig.class;
    }

    @Override
    public Map<String, Class<? extends AuthenticatorRequestHandler<?>>> getAuthenticationRequestHandlerTypes()
    {
        return unmodifiableMap(singletonMap("index", ${pluginName}AuthenticatorRequestHandler.class));
    }
}

File: kotlin-oauth-authenticator-archetype/src/main/resources/archetype-resources/src/main/kotlin/descriptor/__pluginName__AuthenticatorPluginDescriptor.kt
package ${package}.descriptor

import ${package}.authentication.CallbackRequestHandler
import ${package}.authentication.${pluginName}AuthenticatorRequestHandler
import ${package}.config.${pluginName}AuthenticatorPluginConfig
import se.curity.identityserver.sdk.plugin.descriptor.AuthenticatorPluginDescriptor

class ${pluginName}AuthenticatorPluginDescriptor: AuthenticatorPluginDescriptor<${pluginName}AuthenticatorPluginConfig>
{

    override fun getPluginImplementationType() = "${pluginName.toLowerCase()}"

    override fun getConfigurationType() = ${pluginName}AuthenticatorPluginConfig::class.java

    override fun getAuthenticationRequestHandlerTypes() = linkedMapOf(
        "index" to ${pluginName}AuthenticatorRequestHandler::class.java,
        CALLBACK to CallbackRequestHandler::class.java
    )

    companion object
    {
        const val CALLBACK: String = "callback"
    }
}
